src/oemof_tabular_plugins/general/pre_processing.py

```diff
@@ -2,227 +2,341 @@
+                    if not output_parameters_exist:
+                        # if the output_parameters field doesn't exist, add it to the schema
+                        fields.append({
+                            "name": "output_parameters",
+                            "type": "object",
+                            "format": "default"
+                        })
+                        resource_schema["fields"] = fields
+                        resource["schema"] = resource_schema
+
+    # write the updated datapackage.json back to the file
+    with open(datapackage_json_path, "w") as f:
+        json.dump(datapackage, f, indent=4)
+    return
+
+
+def pre_processing_custom_attributes(scenario_dir, element, element_path, element_df, custom_attributes):
+    """Updates the 'output_parameters' field in the CSV file for the specified element if custom
+    attributes are defined.
+
+    :param scenario_dir: scenario directory path
+    :param element: csv filename
+    :param element_path: path of the csv file
+    :param element_df: dataframe containing data from the csv file
+    :param custom_attributes: list of custom attributes included in the model (defined in compute.py)
+    :return: updated 'output_parameters' in csv files and datapackage.json
+    """
+    # iterate over each entry in the dataframe (from csv file)
+    for index, row in element_df.iterrows():
+        # create empty custom attributes dict
+        custom_attributes_dict = {}
+        # set boolean to false
+        has_custom_attributes = False
+        # check if custom_attributes is not none before iterating
+        if custom_attributes is not None:
+            # check if any of the custom attributes list are in the dataframe columns
+            for attribute in custom_attributes:
+                if attribute in element_df.columns:
+                    value = row[attribute]
+                    # add the attribute to the custom attributes dict
+                    custom_attributes_dict[attribute] = value
+                    # set boolean to true
+                    has_custom_attributes = True
+                # if custom attributes are found for this row, add them to 'output_parameters'
+                if has_custom_attributes:
+                    output_parameters_str = json.dumps({"custom_attributes": custom_attributes_dict})
+                    element_df.at[index, 'output_parameters'] = output_parameters_str
+                else:
+                    # no custom attributes found, do not update 'output_parameters'
+                    continue
+    # write the updated dataframe back to the csv file
+    element_df.to_csv(element_path, sep=';', index=False)
+    # if custom attributes were found, update the datapackage.json file
+    if has_custom_attributes:
+        update_datapackage_json_custom_attributes(scenario_dir, element)
+    return
```

Comment:
It returns None, not what the docstring mentions. I would rather remove the mention in the return docstring description